libvast/builtins/query-languages/vastql.cpp

```diff
@@ -43,7 +43,8 @@ class plugin final : public virtual query_language_plugin {
     auto parsed_expr = expression{};
     const auto optional_ws = ignore(*space);
     bool has_expr = true;
-    if (!expr(f, l, parsed_expr)) {
+    const auto expr_parser = optional_ws >> expr;
```

Comment:
```suggestion
    const auto expr_parser = optional_ws >> expr >> optional_ws;
```

Why not that?

Reply:
Because there is further code below this because there may be a pipeline after this, and that already handles the whitespace / eoi at the end. There's no need to handle this twice.

Reply:
Isn't a pipeline optional?

Reply:
What happens if a pipeline is not there?

Reply:
Yes, but the expression as well, and then there's an `optional_ws >> eoi` at the end that we always require.